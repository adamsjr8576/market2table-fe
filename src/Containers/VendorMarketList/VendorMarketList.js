import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './VendorMarketList.scss';
// import { addVendors } from '../../actions';
// import images from '../../images/images';
// import { getAllVendors } from '../../apiCalls';

export const VendorMarketList = () => {

  return (
    <section className='vendor-market-list-section'>
    </section>
  )
}

export default VendorMarketList;
