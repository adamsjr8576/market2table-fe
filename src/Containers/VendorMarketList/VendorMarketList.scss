.vendor-market-list-section {
  width: 45%;
  height: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  background-color: rgba(0,0,0,0.4);
  border: 2px solid khaki;
  border-radius: 20px;
  // background-color: #F2F9F9;
  color: cornsilk;
  letter-spacing: 1px;
}

.back-to-vendors-p {
  font-size: 0.8em;
}

.vendor-market-list-header {
  width: 100%;
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex-wrap: wrap;
  margin-top: 10px;
}

.vendor-market-list-name {
  width: 60%;
  text-align: center;
  margin: 0;
  font-size: 2.5em;
  text-decoration: underline;
}

.vendor-market-list-title {
  width: 100%;
  text-align: center;
  margin: 0;
  font-size: 2em;
}

.vendor-market-list-container {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: column;
  overflow: auto;
  height: 80%;
  border-top: 2px solid khaki;
  background-color: cornsilk;
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 20px;
}

@media screen and (max-width: 1000px) {
  .vendor-market-list-section {
    width: 85%;
    margin-top: 35px;
  }

  .vendor-market-list-header {
    height: 200px;
  }
}
