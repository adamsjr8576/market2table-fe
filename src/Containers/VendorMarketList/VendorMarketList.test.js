import React from 'react';
import { shallow } from 'enzyme';
import { VendorMarketList } from './VendorMarketList';

const mockState = {
  selectedVendor: {name: 'Rolling in the dough', description: 'just a little bakery rolling in some dough', image_link: 'rollthatdough.png'}
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState,
  useDispatch: () => jest.fn()
}));

describe('VendorMarketList', () => {

  let wrapper;
  let mockAddSelectedVendor = jest.fn();
  let mockAddZipCode = jest.fn();
  let mockAddMarkets = jest.fn();
  let mockMarketsLinked;
  let mockSetMarketsLinked = jest.fn();

  it('should match the snapshot with marketsLinked', () => {
    mockMarketsLinked = [{name: 'Golden Farmers Market', id:'456'}, {name: 'Arvada Farmers Market', id:'8372'}];
    wrapper = shallow(<VendorMarketList
      addSelectedVendor={mockAddSelectedVendor}
      addZipCode={mockAddZipCode}
      addMarkets={mockAddMarkets}
      marketsLinked={mockMarketsLinked}
      setMarketsLinked={mockSetMarketsLinked}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot without marketsLinked', () => {
    mockMarketsLinked = [];
    wrapper = shallow(<VendorMarketList
      addSelectedVendor={mockAddSelectedVendor}
      addZipCode={mockAddZipCode}
      addMarkets={mockAddMarkets}
      marketsLinked={mockMarketsLinked}
      setMarketsLinked={mockSetMarketsLinked}
    />);

    expect(wrapper).toMatchSnapshot();
  });

});
