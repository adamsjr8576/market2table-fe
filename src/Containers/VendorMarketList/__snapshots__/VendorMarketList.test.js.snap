// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorMarketList should match the snapshot with marketsLinked 1`] = `
<section
  className="vendor-market-list-section"
>
  <header
    className="vendor-market-list-header"
  >
    <Link
      className="link-back-to-vendors"
      onClick={[Function]}
      to="/vendor/account"
    >
      <img
        alt="icon of reverse array"
        className="undo-image"
        src="undo.png"
      />
      <p
        className="back-to-vendors-p"
      >
        Back To Vendors
      </p>
    </Link>
    <h3
      className="vendor-market-list-name"
    />
    <h4
      className="vendor-market-list-title"
    >
      Markets
    </h4>
  </header>
  <section
    className="vendor-market-list-container"
  >
    <MarketListCard
      id="456"
      key="456"
      marketsLinked={
        Array [
          Object {
            "id": "456",
            "name": "Golden Farmers Market",
          },
          Object {
            "id": "8372",
            "name": "Arvada Farmers Market",
          },
        ]
      }
      name="Golden Farmers Market"
      setMarketsLinked={[MockFunction]}
    />
    <MarketListCard
      id="8372"
      key="8372"
      marketsLinked={
        Array [
          Object {
            "id": "456",
            "name": "Golden Farmers Market",
          },
          Object {
            "id": "8372",
            "name": "Arvada Farmers Market",
          },
        ]
      }
      name="Arvada Farmers Market"
      setMarketsLinked={[MockFunction]}
    />
  </section>
</section>
`;

exports[`VendorMarketList should match the snapshot without marketsLinked 1`] = `
<section
  className="vendor-market-list-section"
>
  <header
    className="vendor-market-list-header"
  >
    <Link
      className="link-back-to-vendors"
      onClick={[Function]}
      to="/vendor/account"
    >
      <img
        alt="icon of reverse array"
        className="undo-image"
        src="undo.png"
      />
      <p
        className="back-to-vendors-p"
      >
        Back To Vendors
      </p>
    </Link>
    <h3
      className="vendor-market-list-name"
    />
    <h4
      className="vendor-market-list-title"
    >
      Markets
    </h4>
  </header>
  <section
    className="vendor-market-list-container"
  >
    <p>
      Please Link your Vendor account to markets
    </p>
  </section>
</section>
`;
